packages/deploy-script-support/src/installInPieces.js

```diff
@@ -1,16 +1,44 @@
+/* global globalThis */
 // @ts-check
 import { E } from '@endo/far';
 import { ZipReader } from '@endo/zip';
 import { encodeBase64, decodeBase64 } from '@endo/base64';
+import { importBundle } from '@endo/import-bundle';
+import crypto from 'crypto';
+
+const computeSha512 = bytes => {
+  const hash = crypto.createHash('sha512');
+  hash.update(bytes);
+  return hash.digest().toString('hex');
+};
+
+const endowments = {
+  console,
+  assert,
+  Base64: globalThis.Base64, // Present only on XSnap
+  URL: globalThis.URL, // Absent only on XSnap
+};
 
 export const installInPieces = async (
   bundle,
   bundler,
-  { maxBytesInFlight = 800_000, log = console.log } = {},
+  { maxBytesInFlight = 800_000, log = console.log, persist = false } = {},
 ) => {
+  // Detect evaluation errors early.
+  await importBundle(bundle, { endowments });
```

Comment:
@michaelfig , this seems to work for me, but when@samsiegart does `yarn deploy:proposal`, he gets the following error. Help? Clues?

```
Bundling bundle maker /home/samsiegart/agoric-sdk/packages/deploy-script-support/src/endo-pieces-contract.js
agoric: (Error#1)
Error#1: VatData unavailable

  at unvailable (.../vat-data/src/index.js:28:29)
  at eval (.../run-protocol/src/vaultFactory/liquidateMinimum.js:105:31)
  at async installInPieces (packages/deploy-script-support/src/installInPieces.js:28:3)
  at async publishRef (packages/deploy-script-support/src/writeCoreProposal.js:77:19)
  at async Promise.all (index 2)
  at async Promise.all (index 2)
  at async Promise.all (index 1)
  at async Promise.all (index 1)

Error#1 ERROR_NOTE: Rejection from: (Error#2) : 24 . 8
Nested error under Error#1
  Error#2: Event: 20.6
  
    at writeCoreProposal (packages/deploy-script-support/src/writeCoreProposal.js:82:28)
    at default (packages/run-protocol/scripts/init-core.js:178:5)
    at async WebSocket.<anonymous> (packages/agoric-cli/src/deploy.js:274:13)
  
error Command failed with exit code 2.
```

Reply:
this is a symptom of calling `importBundle` outside swingset. Let's not, for now.